clients/drcachesim/drpt2trace/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of Google, Inc. nor the names of its contributors may be
+#   used to endorse or promote products derived from this software without
+#   specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+cmake_minimum_required(VERSION 3.7)
+
+include(../../../make/policies.cmake NO_POLICY_SCOPE)
+
+# Add libipt to build system. For kernel traces decoding,
+# we need to activate SIDEBAND, PEVENT and FEATURE_ELF.
+# XXX: libipt does not use the CMake export mechanism for modular separation.
+# It puts us at risk of CMake name collisions. Since libipt's build system is
+# not complicated, we can rewrite a new CMakeList.txt for it and make it modular
+# using the cmake export mechanism.
+set(SIDEBAND ON CACHE BOOL "Enable libipt-sb, a sideband correlation library." FORCE)
+set(PEVENT ON CACHE BOOL "Enable perf_event sideband support." FORCE)
+set(FEATURE_ELF ON CACHE BOOL "Support ELF files." FORCE)
+add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/libipt
+  ${PROJECT_BINARY_DIR}/third_party/libipt)
+
+# drpt2ir depends on libipt and libipt-sb. Also, the head files
+# intel-pt.h and libipt-sb.h are generated into the project binary directory.
+# So we need to add the subdirectories that contain them to the include path.
+include_directories(SYSTEM
+  ${PROJECT_BINARY_DIR}/third_party/libipt/libipt/include
+  ${PROJECT_BINARY_DIR}/third_party/libipt/sideband/include
+)
+
+# libipt.so and libipt-sb.so will be built in the project binary directory.
+# Add link directories for libipt-ext and drpt2ir to find them.
+link_directories(
+  ${PROJECT_BINARY_DIR}/lib
+)
+
+# The library libipt-ext and drpt2ir need to find headers from
+# the following directories.
+include_directories(
+  ${PROJECT_SOURCE_DIR}/third_party/libipt/ptxed/include
+)
+
+# XXX: The repository of lipipt contains some functions that are useful
+# for drpt2ir. But they are not built in libipt.so or libipt-sb.so.
+# So we build them as an external library of libipt and link the external
+# library to drpt2ir. In the future we may need to implement
+# these functions in our library.
```

Comment:
I don't understand the last sentence?  You mean we'd get rid of this ipt-ext library for some reason?

Reply:
My purpose for using libipt-ext is to read the elf part from kcore. Because the functions in libipt-ext don't belong to libipt library and libipt-sb library and only contain 360 lines of c code, I plan to implement a c++ version in the next PR.